SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -259,17 +260,75 @@ void ASpatialDebugger::OnEntityRemoved(const Worker_EntityId EntityId)
 	EntityActorMapping.Remove(EntityId);
 }
 
+void ASpatialDebugger::ActorAuthorityChanged(const Worker_AuthorityChangeOp& AuthOp) const
+{
+	const bool bAuthoritative = AuthOp.authority == WORKER_AUTHORITY_AUTHORITATIVE;
+
+	if (bAuthoritative && AuthOp.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID)
+	{
+		if (NetDriver->VirtualWorkerTranslator == nullptr)
+		{
+			// Currently, there's nothing to display in the debugger other than load balancing information.
+			return;
+		}
+
+		VirtualWorkerId LocalVirtualWorkerId = NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId();
+		FColor LocalVirtualWorkerColor = SpatialGDK::GetColorForWorkerName(NetDriver->VirtualWorkerTranslator->GetLocalPhysicalWorkerName());
+
+		SpatialDebugging* DebuggingInfo = NetDriver->StaticComponentView->GetComponentData<SpatialDebugging>(AuthOp.entity_id);
+		if (DebuggingInfo == nullptr)
+		{
+			// Some entities won't have debug info, so create it now.
+			SpatialDebugging NewDebuggingInfo(LocalVirtualWorkerId, LocalVirtualWorkerColor, SpatialConstants::INVALID_VIRTUAL_WORKER_ID, InvalidServerTintColor, false);
+			Worker_ComponentData DebuggingData = NewDebuggingInfo.CreateSpatialDebuggingData();
+			NetDriver->Connection->SendAddComponent(AuthOp.entity_id, &DebuggingData);
+			return;
+		}
```

Comment:
All entities with the SpatialDebugging component should have been created by the EntityFactory and should have the component data? When would (DebuggingInfo == nullptr) be true?

Reply:
Anything that might be in a snapshot won't have it. Also I'm not sure how it would work if you turned on SpatialDebugging while the system was running, or how it would work over a PiE restart. This just protects us in those cases.